### DISCUSSION TITLE
Nested Wildcards Ignoring Folder Structure

### DISCUSSION THREAD

**Discussion by DrJimmyBrungus** [adieyal/sd-dynamic-prompts#759] [Q&A] [answered]
Not sure if this could be classed as a bug, a feature..

Let's say I have the following file structure - the goal is to have wildcard files describing various outfits for different characters:

```
Characters\

 Jim\
  Casual.txt
   (red t-shirt, jeans)
  Suit.txt
   (black suit, red tie)
  Jim.txt
   __Casual__
   __Suit__

 Bob\
  Casual.txt
   (blue t-shirt, jeans)
  Suit.txt
   (black suit, blue tie)
  Bob.txt
   __Casual__
   __Suit__
```

In my prompt, I enter `__Characters/Jim/Jim__` to pick a random line from within Jim.txt - in this case either Casual or Suit.

What is happening however, is it is picking ANY wildcard named Casual or Suit. It could be Jim, it could be Bob, or it could some other unrelated wildcard somewhere else.

Workaround:

1) Instead of having Jim.txt containing the wildcard lines and promping for `__Characters/Jim/Jim__`, simply prompt for `__Characters/Jim*__` instead. This restricts it to only wilcards found within the Jim folder, as intended.

2) Could ensure every wilcard has a unique name. Eg. Jim-Casual.txt, Bob-Casual.txt

**Comment by akx** [comment, marked as answer]
This should work if `Bob.txt` has

```
__Characters/Bob/Casual__
__Characters/Bob/Suit__
```

See https://github.com/adieyal/dynamicprompts/pull/82 and #377.

**Reply by DrJimmyBrungus** [reply]
Aha, that makes sense. As usual it's something obvious.
